### PR TITLE
lldb: support new rank device name

### PR DESCRIPTION
**Pull request by dlagneux-upmem**
Regions are made up of a single rank, so rank ID is always 0. We decided to get rid of the region ID, which is replaced by the rank ID.

Tell me if you prefer checking explicitely that the device node exists (with the access call), or you prefer updating the failure message at rank allocation (telling that module might be obsolete).

And I do not know how you format the code?

### CODE REVIEW

File: lldb/scripts/dpu/dpu_commands.py [hunk 1]
```diff
@@ -58,7 +57,7 @@ def get_dpu_from_command(command, debugger, target):
         "dpu", "(struct dpu_t *)" + str(addr))
 
 
-def get_region_id_and_rank_id(rank, target):
+def get_rank_id(rank, target):
```
Comment (ghost) [inline lldb/scripts/dpu/dpu_commands.py:60]: This is find and does exactly what was done before but adapt for the new driver.
But we might want to change it as now the rank_id exposed in the `struct dpu_rank_t` is the real rank_id. We just need to do something clean to detect backend that are not hardware.I think, I'll do it after you will have push that.

**Comment by ghost** [comment]
You also need to modify `<backends>/host-lldb-attach-dpu/lldb-attach-dpu.c`

**Comment by ghost** [comment]
For formatting python:
- use `autopep8 -a -a -i <python_file>` (But the file is already not well formatted...)

For C++:
- use `clang-format -i <c++_file>`

File: lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp [hunk 2]
```diff
@@ -140,15 +140,21 @@ ProcessDpu::Factory::Attach(
...
   dpu_id = pid % 100;
   slice_id = (pid / 100) % 100;
   rank_id = (pid / (100 * 100)) % 100;
-  region_id = (pid / (100 * 100 * 100)) % 100;
+
+  char rank_path[128];
+  sprintf(rank_path, "/dev/dpu_rank%u", rank_id);
+
+  if (access(rank_path, F_OK) != 0) {
+    return Status("%s does not exist. The dpu module might be outdated ", rank_path)
```
Comment (ghost) [inline lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp]:
```suggestion
    return Status("%s does not exist. Run 'dpu-diag' for a more full diagnostic ", rank_path)
```

**Comment by ghost** [comment]
also, please prefix your commit message with `dpu: `

**Comment by ghost** [comment]
Your push force did not change the commit message

**Comment by dlagneux-upmem** [comment]
Daamn, I misread your message, I tought you talked about the Status message...

**Comment by ghost** [comment]
I prefer the previous status message ^^ maybe you could even add some quote around `%s` (`'%s'`)

**Review by ghost — APPROVED** [review]

**Pull request merged**